Create DrawGrid.json

TheGemDev opened the pull request:
draw a virtual grid.

<img width="812" alt="image" src="https://github.com/GDevelopApp/GDevelop-extensions/assets/53819287/5e3b89d0-81d5-489c-af93-e638f46b4ebf">

<img width="832" alt="image" src="https://github.com/GDevelopApp/GDevelop-extensions/assets/53819287/d50eba86-c5c8-4c3e-b254-4b38cb9eea00">


<img width="566" alt="image" src="https://github.com/GDevelopApp/GDevelop-extensions/assets/53819287/2cd824cf-b6cf-4cde-a89e-b02efe318d41">

TheGemDev commented:
Project - [Draw Grid.zip](https://github.com/GDevelopApp/GDevelop-extensions/files/14323360/Draw.Grid.zip)

Extension - 
[Draw Grid Extension.zip](https://github.com/GDevelopApp/GDevelop-extensions/files/14323371/DrawGrid.zip)

Oxey405 reviewed (changes requested):
So it's a great idea but there are a few things to fix I think before this gets published
1. Performance
Performance is really bad because of how you draw the cells (using a loop)
At least you should warn people that it's gonna take a long time to render for fine-grain loops
2. Coloring
I think it would also be great to explain how to change the color of the grid for example in the description of the extension, the action or in a help page

Other than that it looks great :)

Oxey405 commented:
> Project - [Draw Grid.zip](https://github.com/GDevelopApp/GDevelop-extensions/files/14323360/Draw.Grid.zip)
> 
> Extension - [Draw Grid Extension.zip](https://github.com/GDevelopApp/GDevelop-extensions/files/14323371/DrawGrid.zip)

Thanks

D8H commented:
> Performance is really bad because of how you draw the cells (using a loop)

The issue is not the loop. It's the instance creation.

TheGemDev commented:
fixed the performance and added a way to change colors in the action. Also, if no instance of the shape painter object exists,  the action creates one.

<img width="566" alt="image" src="https://github.com/GDevelopApp/GDevelop-extensions/assets/53819287/d0046aeb-d4d8-4491-a59a-ba2bd6598b9a">

D8H commented:
What happens when the camera is moved?
Can extension users use the grid to debug path finding of a game with scrolling?
What do you think about making a behavior that updates the grid when needed?
